### ISSUE TITLE
Iterable should contain some

### ISSUE THREAD

**Issue by goreRatzete** [MarkusAmshove/Kluent#59]
Currently testing some iterables and found that Kluent lacks support for the following:

1) I have a List which contains some elements that are expected.
2) I have a List which contains actual elements

2 should contain at least some of 1, but not necessarily all (shouldContainAll).

It's kind of the inverse of shouldNotContainAny. 
-> should contain some

```
val cities = listOf("Israel", "Phoenix", "Egypt")
val actual = listOf("Israel", "Some", "other", "Phoenix") // Egypt is missing

// long version
actual.any { it in cities }.shouldBeTrue()

// fails, not the one i am looking for
actual shouldContainAll cities

// what I propose
actual shouldContainSome cities
```

**Comment by goreRatzete** [comment]
I would also propose the same for CharSequence.

Here we are both missing `shouldNotContainAny` and `shouldContainSome`.

**Comment by MarkusAmshove** [comment]
This is a nice idea.

Would the threshold for `Some` be 1? So everytime at least one item is found the test passes?

**Comment by MarkusAmshove** [comment]
This is how I understand it:

![image](https://user-images.githubusercontent.com/2401875/30238744-900efc02-954d-11e7-8e9a-3e550fd72759.png)

Also trying to figure out the correct failure message up front now :)

**Comment by goreRatzete** [comment]
Yes, exactly.

We expect some of the elements to be there, but we might not know which yet. 

Error message looks fine.

**Comment by MarkusAmshove** [comment]
It is now included in 1.29 :)
